resources/opdb/resource_database.go

```diff
@@ -0,0 +1,175 @@
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package opdb
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/opdb/client/operations"
+	opdbmodels "github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/opdb/models"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+var (
+	_ resource.Resource = &databaseResource{}
+)
+
+type databaseResource struct {
+	client *cdp.Client
+}
+
+func NewDatabaseResource() resource.Resource {
+	return &databaseResource{}
+}
+
+func (r *databaseResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_operational_database"
+}
+
+func (r *databaseResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client = utils.GetCdpClientForResource(req, resp)
+}
+
+func (r *databaseResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "OPDB cluster creation process requested.")
+	var data databaseResourceModel
+	tflog.Info(ctx, fmt.Sprintf("Creating OPDB with name: %s", data.DatabaseName.ValueString()))
+	diags := req.Plan.Get(ctx, &data)
+	tflog.Info(ctx, fmt.Sprintf("OPDB resource model: %+v", data))
+	resp.Diagnostics.Append(diags...)
+	tflog.Info(ctx, fmt.Sprintf("Diags: %+v", resp.Diagnostics))
+	if resp.Diagnostics.HasError() {
+		tflog.Warn(ctx, "OPDB resource model has error, stopping the creation process.")
+		return
+	}
+
+	params := operations.NewCreateDatabaseParamsWithContext(ctx)
+	params.WithInput(fromModelToDatabaseRequest(data, ctx))
+
+	tflog.Info(ctx, fmt.Sprintf("Sending create request for OPDB with name: %s", data.DatabaseName.ValueString()))
+	res, err := r.client.Opdb.Operations.CreateDatabase(params)
+
+	tflog.Info(ctx, fmt.Sprintf("Create request for OPDB with name: %s has been sent with the result of: %+v", data.DatabaseName.ValueString(), res))
+	if err != nil {
+		utils.AddDatabaseDiagnosticsError(err, &resp.Diagnostics, "create OPDB")
+		return
+	}
+
+	data.DatabaseName = types.StringPointerValue(res.Payload.DatabaseDetails.DatabaseName)
+	data.Crn = types.StringPointerValue(res.Payload.DatabaseDetails.Crn)
+	data.Environment = types.StringValue(res.Payload.DatabaseDetails.EnvironmentName)
+	data.Status = types.StringValue(string(res.Payload.DatabaseDetails.Status))
+
+	data.ScaleType = types.StringValue(string(res.Payload.DatabaseDetails.ScaleType))
+	data.StorageLocation = types.StringValue(res.Payload.DatabaseDetails.StorageLocation)
```

Comment:
this, and the same fill part in the `Read` looks kinda identical. If possible could you please extract them into a single function?